pkg/apply/apply.go

```diff
@@ -90,11 +90,11 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 	if err := MergeObjectForUpdate(existing, obj); err != nil {
 		return errors.Wrapf(err, "could not merge object %s with existing", objDesc)
 	}
-	if !equality.Semantic.DeepDerivative(obj, existing) {
+	if !equality.Semantic.DeepEqual(existing, obj) {
```

Comment:
this will always return false and you will update the object.

as an example: the object's resourceVersion is unset for objects loaded from  manifests.
did you notice a case where the log at L97 was printed ?

can you confirm that you have properly tested it on a system and you noticed no adverse effects ?
(like sriov device plugin restarting)